src/v/cloud_storage/remote_partition.cc

```diff
@@ -272,6 +273,13 @@ class partition_record_batch_reader_impl final
         }
 
         auto it = seek_segment(config);
+
+        if (it == _partition->end()) {
```

Comment:
@andrwng took a look at what happens we bail out here (as we already do in a couple of cases above), basically we query the local partitions instead. 

I guess this scenario can occur when the query is close to the dividing line between the remote and local segments: if the time query is for a point before the oldest local segment, we go down this path, but we can still return end() if the point is newer than the newest remote segment.